### PR TITLE
Fix for NonprofitsController#update causing infinite loop

### LINKED ISSUE #1104: [BUG] Setting "Transaction Message" fails with stack overflow
When I visit Settings, Transaction Message and enter a value and save, the XHR request hangs (no response) and Houdini begins to consume 100% CPU. Console shows:

```
   (0.9ms)  SELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC
* Locale set to 'en'
  User Load (0.8ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/concerns/controllers/user/authorization.rb:71:in `administered_nonprofit'
   (1.2ms)  SELECT host_id FROM roles WHERE ("user_id" IN (1)) AND (roles.name IN ($Q$nonprofit_admin$Q$, $Q$nonprofit_associate$Q$))
  ↳ app/legacy_lib/query_roles.rb:19:in `host_ids'
  Nonprofit Load (1.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 ORDER BY "nonprofits"."id" DESC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/concerns/controllers/user/authorization.rb:73:in `administered_nonprofit'
  Nonprofit Load (0.5ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/legacy_lib/fetch_nonprofit.rb:10:in `with_params'
   (0.6ms)  SELECT host_id FROM roles WHERE ("user_id" IN (1)) AND (roles.name IN ($Q$nonprofit_admin$Q$, $Q$nonprofit_associate$Q$))
  ↳ app/legacy_lib/query_roles.rb:19:in `host_ids'
  CACHE Nonprofit Load (0.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 ORDER BY "nonprofits"."id" DESC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/concerns/controllers/user/authorization.rb:73:in `administered_nonprofit'
  CACHE Nonprofit Load (0.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/legacy_lib/fetch_nonprofit.rb:10:in `with_params'
   (0.8ms)  SELECT COUNT(roles) FROM roles WHERE (name IN ($Q$nonprofit_admin$Q$, $Q$nonprofit_associate$Q$)) AND ("user_id" IN (1)) AND ("host_id" IN (1))
  ↳ app/legacy_lib/query_roles.rb:11:in `user_has_role?'
   (0.4ms)  SELECT host_id FROM roles WHERE ("user_id" IN (1)) AND (roles.name IN ($Q$nonprofit_admin$Q$, $Q$nonprofit_associate$Q$))
  ↳ app/legacy_lib/query_roles.rb:19:in `host_ids'
  CACHE Nonprofit Load (0.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 ORDER BY "nonprofits"."id" DESC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/concerns/controllers/user/authorization.rb:73:in `administered_nonprofit'
  CACHE Nonprofit Load (0.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/legacy_lib/fetch_nonprofit.rb:10:in `with_params'
  TRANSACTION (0.3ms)  BEGIN
  ↳ app/controllers/nonprofits_controller.rb:61:in `update'
  Nonprofit Exists? (0.6ms)  SELECT 1 AS one FROM "nonprofits" WHERE "nonprofits"."slug" = $1 AND "nonprofits"."id" != $2 AND "nonprofits"."city_slug" = $3 AND "nonprofits"."state_code_slug" = $4 LIMIT $5  [["slug", "software-freedom-conservancy"], ["id", 1], ["city_slug", "brooklyn"], ["state_code_slug", "ny"], ["LIMIT", 1]]
  ↳ app/controllers/nonprofits_controller.rb:61:in `update'
  ActiveStorage::Attachment Load (0.5ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 1], ["record_type", "Nonprofit"], ["name", "logo"], ["LIMIT", 1]]
  ↳ app/models/concerns/image/attachment_extensions.rb:43:in `block in has_one_attached_with_default'
  ActiveStorage::Attachment Load (0.4ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 1], ["record_type", "Nonprofit"], ["name", "background_image"], ["LIMIT", 1]]
  ↳ app/models/concerns/image/attachment_extensions.rb:43:in `block in has_one_attached_with_default'
  ActiveStorage::Attachment Load (0.5ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 1], ["record_type", "Nonprofit"], ["name", "main_image"], ["LIMIT", 1]]
  ↳ app/models/concerns/image/attachment_extensions.rb:43:in `block in has_one_attached_with_default'
  ActiveStorage::Attachment Load (0.5ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 1], ["record_type", "Nonprofit"], ["name", "second_image"], ["LIMIT", 1]]
  ↳ app/models/concerns/image/attachment_extensions.rb:43:in `block in has_one_attached_with_default'
  ActiveStorage::Attachment Load (0.4ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 1], ["record_type", "Nonprofit"], ["name", "third_image"], ["LIMIT", 1]]
  ↳ app/models/concerns/image/attachment_extensions.rb:43:in `block in has_one_attached_with_default'
  TRANSACTION (0.2ms)  COMMIT
  ↳ app/controllers/nonprofits_controller.rb:61:in `update'
   (0.5ms)  SELECT host_id FROM roles WHERE ("user_id" IN (1)) AND (roles.name IN ($Q$nonprofit_admin$Q$, $Q$nonprofit_associate$Q$))
  ↳ app/legacy_lib/query_roles.rb:19:in `host_ids'
  CACHE Nonprofit Load (0.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 ORDER BY "nonprofits"."id" DESC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/controllers/concerns/controllers/user/authorization.rb:73:in `administered_nonprofit'
  CACHE Nonprofit Load (0.0ms)  SELECT "nonprofits".* FROM "nonprofits" WHERE "nonprofits"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/legacy_lib/fetch_nonprofit.rb:10:in `with_params'
  CACHE Nonprofit Exists? (0.0ms)  SELECT 1 AS one FROM "nonprofits" WHERE "nonprofits"."slug" = $1 AND "nonprofits"."id" != $2 AND "nonprofits"."city_slug" = $3 AND "nonprofits"."state_code_slug" = $4 LIMIT $5  [["slug", "software-freedom-conservancy"], ["id", 1], ["city_slug", "brooklyn"], ["state_code_slug", "ny"], ["LIMIT", 1]]
  ↳ app/controllers/application_controller.rb:25:in `json_saved'
  CACHE ActiveStorage::Attachment Load (0.0ms)  SELECT "active_storage_attachments".* FROM "active_storage_attachments" WHERE "active_storage_attachments"."record_id" = $1 AND "active_storage_attachments"."record_type" = $2 AND "active_storage_attachments"."name" = $3 LIMIT $4  [["record_id", 1], ["record_type", "Nonprofit"], ["name", "main_image"], ["LIMIT", 1]]
  ↳ app/models/nonprofit.rb:179:in `as_json'
  ActiveStorage::Blob Load (0.5ms)  SELECT "active_storage_blobs".* FROM "active_storage_blobs" WHERE "active_storage_blobs"."id" = $1 LIMIT $2  [["id", 4], ["LIMIT", 1]]
  ↳ app/models/nonprofit.rb:179:in `as_json'
method=PUT path=/nonprofits/1.json format=json controller=NonprofitsController action=update status=500 error='SystemStackError: stack level too deep' duration=477.68 view=0.00 db=33.98 time=30768.38 exception=["SystemStackError", "stack level too deep"] exception_object=stack level too deep
  
SystemStackError (stack level too deep):
  
app/models/nonprofit.rb:179:in `as_json'
app/models/nonprofit.rb:179:in `as_json'
... (repeated a few hundred times)
app/models/nonprofit.rb:179:in `as_json'
app/controllers/application_controller.rb:27:in `json_saved'
app/controllers/nonprofits_controller.rb:62:in `update'
app/controllers/concerns/controllers/locale.rb:22:in `switch_locale'
```

### PR DESCRIPTION
**Pull request by wwahammy**
Depends on #1111

Closes #1104

### CODE REVIEW

File: app/views/nonprofits/show.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    :email,
+    :website,
+    :summary,
+    :categories,
+    :full_description,
+    :published,
+    :no_anon,
+    :timezone,
+    :brand_color,
+    :hide_activity_feed
```
Comment (BenSturmfels) [inline app/views/nonprofits/show.json.jbuilder:20]: It looks like this is missing a comma on the end of the line

File: app/views/nonprofits/show.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    :state_code, 
+    :state_code_slug, 
+    :city, 
+    :city_slug,
+    :slug,
+    :country,
```
Comment (BenSturmfels) [inline app/views/nonprofits/show.json.jbuilder:26]: After adding the above comma, I'm getting `ActionView::Template::Error (undefined method `country' for #<Nonprofit:0x00007fda297f6ad8>)`. Removing the `:country` symbol fixes this. Looking at `db/schema.rb` it appears that there's no such field on the `nonprofits` model. No big deal for my use-case, but might be worth considering in the future.

**Review by BenSturmfels — CHANGES_REQUESTED** [review]
With the above changes, this works fine. Thanks!

**Review by clarissalimab — APPROVED** [review]
It looks good to me, but I think we should fix what Ben found before merging!

**Pull request merged**